issue by liuzwZYW

> [@liuzwZYW](https://github.com/liuzwZYW) your case seems to be a different issue. The log shows `extensions: [".js", ".json", ".node"]`.[@liuzwZYW](https://github.com/liuzwZYW) 你的情况似乎是另一个问题。日志显示 `extensions: [".js", ".json", ".node"]`。
> 
> oh, that my new test log:
> 
> ```sh
> + pwd
> /data/jenkins/workspace/ithinkdt-portal-front
> + ls -al ./eslint.config.js ./ocx-test.js ./postcss.config.js ./stylelint.config.js
> -rw-r--r--    1 root     root          5648 Oct 13 01:41 ./eslint.config.js
> -rw-r--r--    1 root     root            29 Nov 27 01:19 ./ocx-test.js
> -rw-r--r--    1 root     root           424 Nov  3 05:47 ./postcss.config.js
> -rw-r--r--    1 root     root           145 Oct 13 01:41 ./stylelint.config.js
> + OXC_LOG=DEBUG node ./test.mjs
> directory /data/jenkins/workspace/ithinkdt-portal-front
> 2025-11-27T01:22:07.035889Z DEBUG resolve{path="/data/jenkins/workspace/ithinkdt-portal-front" specifier="/data/jenkins/workspace/ithinkdt-portal-front/ocx-test.js"}: oxc_resolver: options=ResolveOptions { cwd: None, tsconfig: None, alias: [], alias_fields: [], condition_names: [], enforce_extension: Disabled, exports_fields: [["exports"]], imports_fields: [["imports"]], extension_alias: [], extensions: [".js", ".json", ".node"], fallback: [], fully_specified: false, main_fields: ["main"], main_files: ["index"], modules: ["node_modules"], resolve_to_context: false, prefer_relative: false, prefer_absolute: false, restrictions: [], roots: [], symlinks: true, builtin_modules: false, module_type: false, allow_package_exports_in_directory_resolve: false, yarn_pnp: false } path="/data/jenkins/workspace/ithinkdt-portal-front" specifier="/data/jenkins/workspace/ithinkdt-portal-front/ocx-test.js" err=NotFound("/data/jenkins/workspace/ithinkdt-portal-front/ocx-test.js")
> specifier /data/jenkins/workspace/ithinkdt-portal-front/ocx-test.js
> resolution {
>   error: "Cannot find module '/data/jenkins/workspace/ithinkdt-portal-front/ocx-test.js'"
> }
> 2025-11-27T01:22:07.036841Z DEBUG resolve{path="/data/jenkins/workspace/ithinkdt-portal-front" specifier="ocx-test.js"}: oxc_resolver: options=ResolveOptions { cwd: None, tsconfig: None, alias: [], alias_fields: [], condition_names: [], enforce_extension: Disabled, exports_fields: [["exports"]], imports_fields: [["imports"]], extension_alias: [], extensions: [".js", ".json", ".node"], fallback: [], fully_specified: false, main_fields: ["main"], main_files: ["index"], modules: ["node_modules"], resolve_to_context: false, prefer_relative: false, prefer_absolute: false, restrictions: [], roots: [], symlinks: true, builtin_modules: false, module_type: false, allow_package_exports_in_directory_resolve: false, yarn_pnp: false } path="/data/jenkins/workspace/ithinkdt-portal-front" specifier="ocx-test.js" err=NotFound("ocx-test.js")
> specifier ocx-test.js
> resolution { error: "Cannot find module 'ocx-test.js'" }
> 2025-11-27T01:22:07.037184Z DEBUG resolve{path="/data/jenkins/workspace/ithinkdt-portal-front" specifier="./ocx-test.js"}: oxc_resolver: options=ResolveOptions { cwd: None, tsconfig: None, alias: [], alias_fields: [], condition_names: [], enforce_extension: Disabled, exports_fields: [["exports"]], imports_fields: [["imports"]], extension_alias: [], extensions: [".js", ".json", ".node"], fallback: [], fully_specified: false, main_fields: ["main"], main_files: ["index"], modules: ["node_modules"], resolve_to_context: false, prefer_relative: false, prefer_absolute: false, restrictions: [], roots: [], symlinks: true, builtin_modules: false, module_type: false, allow_package_exports_in_directory_resolve: false, yarn_pnp: false } path="/data/jenkins/workspace/ithinkdt-portal-front" specifier="./ocx-test.js" err=NotFound("./ocx-test.js")
> specifier ./ocx-test.js
> resolution { error: "Cannot find module './ocx-test.js'" }
> ``` 

 _Originally posted by @liuzwZYW in [#824](https://github.com/oxc-project/oxc-resolver/issues/824#issuecomment-3583856196)_

--- a/src/file_system.rs
+++ b/src/file_system.rs
@@ -173,9 +173,17 @@ impl FileSystemOs {
                 Ok(result.into())
             } else if #[cfg(target_os = "linux")] {
                 use rustix::fs::{AtFlags, CWD, FileType, StatxFlags};
-                let statx = rustix::fs::statx(CWD, path, AtFlags::STATX_DONT_SYNC, StatxFlags::TYPE)?;
-                let file_type = FileType::from_raw_mode(statx.stx_mode.into());
-                Ok(FileMetadata::new(file_type.is_file(), file_type.is_dir(), file_type.is_symlink()))
+                match rustix::fs::statx(CWD, path, AtFlags::STATX_DONT_SYNC, StatxFlags::TYPE) {
+                    Ok(statx) => {
+                        let file_type = FileType::from_raw_mode(statx.stx_mode.into());
+                        Ok(FileMetadata::new(file_type.is_file(), file_type.is_dir(), file_type.is_symlink()))
+                    }
+                    Err(rustix::io::Errno::NOSYS) => {
+                        // statx is not available (kernel < 4.11), fall back to fs::metadata
+                        fs::metadata(path).map(FileMetadata::from)
+                    }
+                    Err(err) => Err(err.into()),
+                }
             } else {
                 fs::metadata(path).map(FileMetadata::from)
             }
@@ -192,9 +200,17 @@ impl FileSystemOs {
                 Ok(crate::windows::symlink_metadata(path)?.into())
             } else if #[cfg(target_os = "linux")] {
                 use rustix::fs::{AtFlags, CWD, FileType, StatxFlags};
-                let statx = rustix::fs::statx(CWD, path, AtFlags::SYMLINK_NOFOLLOW, StatxFlags::TYPE)?;
-                let file_type = FileType::from_raw_mode(statx.stx_mode.into());
-                Ok(FileMetadata::new(file_type.is_file(), file_type.is_dir(), file_type.is_symlink()))
+                match rustix::fs::statx(CWD, path, AtFlags::SYMLINK_NOFOLLOW, StatxFlags::TYPE) {
+                    Ok(statx) => {
+                        let file_type = FileType::from_raw_mode(statx.stx_mode.into());
+                        Ok(FileMetadata::new(file_type.is_file(), file_type.is_dir(), file_type.is_symlink()))
+                    }
+                    Err(rustix::io::Errno::NOSYS) => {
+                        // statx is not available (kernel < 4.11), fall back to fs::symlink_metadata
+                        fs::symlink_metadata(path).map(FileMetadata::from)
+                    }
+                    Err(err) => Err(err.into()),
+                }
             } else {
                 fs::symlink_metadata(path).map(FileMetadata::from)
             }

